BenchPress/Helpers/BenchPress.Azure/Common.psm1

```diff
@@ -255,11 +255,22 @@ function Confirm-Resource {
   } elseif ($ConfirmResult.Success) {
     if ($PropertyKey) {
       $ActualValue = $ConfirmResult.ResourceDetails
-      $Keys = $PropertyKey.Split(".")
+      $Keys = ($PropertyKey -split '[\[\]\.]').Where({ $_ -ne "" })
```

Comment:
Do you mind adding a comment or two about some of these lines?

Specifically 258 and a little about the for-loop block below.